astropy/wcs/src/unit_list_proxy.c

```diff
@@ -91,8 +91,11 @@ PyUnitListProxy_New(
   if (units_dict == NULL) {
     return NULL;
   }
-
+#if PY_VERSION_HEX >= 0x030d00c1
+  unit_class = PyDict_GetItemStringRef(units_dict, "Unit");
```

Comment:
This is not OK: since it is now a strong reference, you need to `Py_DECREF` it when you are done.

Reply:
oh my, of course 🤦🏻‍♂️ 
Thanks !